splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceDateFormatter.java

```diff
@@ -14,29 +14,88 @@
+        // is "yyyy-M-d".
+        if (format == null) {
+            format = "yyyy-M-d";
+            this.format = "yyyy-MM-dd";
+        }
+        else {
+            this.format = format;
+
+            // The valid format characters of DateTimeFormatter documented at the following link
+            // can be used to identify separator characters like '-' or '/' :
+            // See https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+
+            // Any non-format character can be a separator character.
+            // This pattern matches one or more non-format characters.
+            String
+            excludeFormatCharsPattern = "([^GuyDMLdQqYwWEecFahKkHmsSAnNVzOXxZp]+)";
+
+            // The string to replace ("MM" or "dd") may occur at the beginning or
+            // end of the format string, with no characters preceding it
+            // or following it.
+
+            // We must not replace "MM" or "dd" if its neighbor is a
+            // format character.
+            // This pattern excludeFormatCharsPattern or the empty string
+            // must be present on both sides of the match string.
+
             // Replace MM with M in the date format string.
-            String pattern = "([^M]*)([M])([M])([^M]*)";
-            this.format = this.format.replaceFirst(pattern, "$1$2$4");
-    
+            String matchPattern;
+
+            matchPattern = format("(^[M])([M])%s", excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3");
+            matchPattern = format("%s([M])([M]$)", excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3");
+            matchPattern = format("%s([M])([M])%s", excludeFormatCharsPattern,
+                                                    excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3$4");
+
             // Replace dd with d in the date format string.
-            pattern = "([^d]*)([d])([d])([^d]*)";
-            this.format = this.format.replaceFirst(pattern, "$1$2$4");
+            matchPattern = format("(^[d])([d])%s", excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3");
+            matchPattern = format("%s([d])([d]$)", excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3");
+            matchPattern = format("%s([d])([d])%s", excludeFormatCharsPattern,
+                                                    excludeFormatCharsPattern);
+            format = format.replaceFirst(matchPattern, "$1$3$4");
+        }
+        try {
+            this.formatter =
+                java.time.format.DateTimeFormatter.ofPattern(format);
+        }
+        catch (IllegalArgumentException e) {
+            // If the format is bad, this.formatter will be null, and the next time
+            // getFormatter() is called we'll throw a SQLException.
```

Comment:
We'd rather print out a WARN message to the logs.

Reply:
Thanks.. it seems the splice way of relaying syntax errors like this in the user's SQL is to throw a StandardException or a SQLException.  The constructor can't throw the exception, so the first time getformatter is called the exception will be thrown, and a warning such as the following will appear in the logs:

`2019-07-14 20:54:13,369 (DRDAConnThread_567) WARN  [c.s.p.Exceptions] - Parsing exception, original stacktrace:
java.lang.RuntimeException: ERROR 38000: The exception 'java.sql.SQLException: Error parsing datetime with pattern: yyyy-MM-dddd. Try using an ISO8601 pattern such as, yyyy-MM-dd'T'HH:mm:ss.SSSZZ, yyyy-MM-dd'T'HH:mm:ssZ or yyyy-MM-dd' was thrown while evaluating an expression.
`